### PR TITLE
[WB-6863] Fix `from wandb import magic` to not require tensorflow

### PR DESCRIPTION
**Pull request by raubitsj**
Fixes WB-6863

Description
-----------
Slight tweaks to https://github.com/wandb/client/pull/2944.  Instead of aborting if tensorflow isnt installed the feature will still do other things (log absl / argparse / sys args)

Testing
-------
Added a yea test when there is no tensorflow

### CODE REVIEW

File: wandb/integration/magic.py [hunk 1]
```diff
@@ -504,7 +506,7 @@ def magic_install(init_args=None):
     # process system args
     _process_system_args()
     # install argparse wrapper
-    in_jupyter_or_ipython = wandb.wandb_sdk.lib.ipython._get_python_type != "python"
+    in_jupyter_or_ipython = wandb.wandb_sdk.lib.ipython._get_python_type() != "python"
```
Comment (kptkin) [inline wandb/integration/magic.py:509]: oops... 😅

File: functional_tests/magic/02-no-keras.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  shard: noml
+command:
+    args:
+        - --layers=5
+assert:
+  - :wandb:runs_len: 1
+  - :wandb:runs[0][config][epochs]: 4
+  - :wandb:runs[0][config][layers]: 5
+  - :wandb:runs[0][exitcode]: 0
+"""
```
Comment (kptkin) [inline functional_tests/magic/02-no-keras.py:18]: is there anything else `magic` does if no ml is installed?

File: wandb/integration/magic.py [hunk 3]
```diff
@@ -443,7 +444,9 @@ def _magic_update_config():
     if wandb.run and wandb.run.config:
         c = wandb.run.config
         user_config = dict(c.items())
-        if user_config:
+        # ignore keys set by magic integration when checking
+        # if user added any keys
+        if set(user_config).difference({"magic"}):
```
Comment (kptkin) [inline wandb/integration/magic.py:449]: nice

**Review by kptkin — APPROVED** [review]
Thanks! Nice improvement for this magic.

File: functional_tests/magic/02-no-keras.py (hunk 2 above)
Comment (raubitsj) [reply functional_tests/magic/02-no-keras.py:18]: i guess i could check stdout/stderr.  but that is it.   Will add more magic tests when I get time, though looking at this code it is clear it needs some serious cleanups (and annotations!) :)

Comment (kptkin) [reply functional_tests/magic/02-no-keras.py:18]: yeah... I was looking at this code and i think it needs a lot of love. 
Also I see that we are using `global` a lot in this file.

**Pull request merged**